pkg/config/config.go

```diff
@@ -37,38 +37,63 @@ var (
 	validRoles = []string{"controlplane", "node"}
 )
 
-type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+// Top level config
+type Config struct {
```

Comment:
Should we add a metadata section so we can be explicit about versioning, too? What's our standard practice with other files like this?

Reply:
Can we put off that decision until we find ourselves needing to make a breaking change?

Reply:
We could. Would we get any benefit from declaring this a beta or even alpha version?